src/gui/qgsexpressionbuilderwidget.cpp

```diff
@@ -422,6 +424,35 @@ void QgsExpressionBuilderWidget::btnNewFile_pressed()
   }
 }
 
+void QgsExpressionBuilderWidget::btnRemoveFile_pressed()
+{
+  int currentRow = cmbFileNames->currentRow();
+  QString fileName = cmbFileNames->currentItem()->text();
+  if ( QFile::remove( mFunctionsPath + QDir::separator() + fileName.append( ".py" ) ) )
+  {
+    {
+      const QSignalBlocker blocker( cmbFileNames );
+      QListWidgetItem *itemToRemove = cmbFileNames->takeItem( currentRow );
```

Comment:
```suggestion
      QListWidgetItem *itemToRemove = whileBlocking( cmbFileNames )->takeItem( currentRow );
```

Then you don't need QSignalBlocker!

Reply:
Done, thanks!